From 4a64c4bf048f8089e82a30c552a95bbe0659d15f Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Thu, 12 Nov 2015 15:40:00 -0800
Subject: [PATCH] nexus/build/os/linuxkernel/os_flags.inc: don't add
 NEXUS_EXTRA_CFLAGS to NEXUS_CFLAGS

NEXUS_EXTRA_CFLAGS is used to pass --sysroot=XXX when compiling for user space.
Since we don't want to include the user space sysroot when compiling for kernel
space, prevent NEXUS_EXTRA_CFLAGS being added to NEXUS_CFLAGS here and add
NEXUS_EXTRA_KERNEL_CFLAGS instead.

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 nexus/build/os/linuxkernel/os_flags.inc | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/nexus/build/os/linuxkernel/os_flags.inc b/nexus/build/os/linuxkernel/os_flags.inc
index 1f2d036..244fd10 100644
--- a/nexus/build/os/linuxkernel/os_flags.inc
+++ b/nexus/build/os/linuxkernel/os_flags.inc
@@ -196,5 +196,8 @@ endif
 # Flag to generate dependencies
 CDEP_FLAG = -MMD
 
-# NEXUS_EXTRA_CFLAGS can be defined by the user to bring in extra CFLAGS
-NEXUS_CFLAGS += ${NEXUS_EXTRA_CFLAGS}
+# NEXUS_EXTRA_CFLAGS is used to pass --sysroot=XXX when compiling for user space.
+# Since we don't want to include the user space sysroot when compiling for kernel
+# space, prevent NEXUS_EXTRA_CFLAGS being added to NEXUS_CFLAGS here and add
+# NEXUS_EXTRA_KERNEL_CFLAGS instead.
+NEXUS_CFLAGS += ${NEXUS_EXTRA_KERNEL_CFLAGS}
-- 
1.9.1

